Chap_API_Server.tex

```diff
@@ -238,6 +237,251 @@ \subsection{\code{PMIx_server_register_nspace}}
 The number of local processes for any given namespace is generally fixed at the time of application launch. Calls to \refapi{PMIx_Spawn} result in processes launched in their own namespace, not that of their parent. However, it is possible for processes to \textit{migrate} to another node via a call to \refapi{PMIx_Job_control_nb}, thus resulting in a change to the number of local processes on both the initial node and the node to which the process moved. It is therefore \textit{critical} that applications not migrate processes without first ensuring that \ac{PMIx}-based collective operations are not in progress, and that no such operations be initiated until process migration has completed.
 \adviceuserend
 
+%%%%%%%%%%%
+\subsubsection{Assembling the registration information}
+\label{chap:api_server:assemble}
+
+The following description is not intended to represent the actual layout of information in a given \ac{PMIx} library. Instead, it is describes how information provided in the \refarg{info} parameter of the \refapi{PMIx_server_register_nspace} shall be organized for proper processing by a \ac{PMIx} server library. The ordering of the various information elements is arbitrary - they are presented in a top-down hierarchical form solely for clarity in reading.
+
+\advicermstart
+Creating the \refarg{info} array of data requires knowing in advance the number of elements required for the array. This can be difficult to compute and somewhat fragile in practice. One method for resolving the problem is to create a linked list of objects, each containing a single \refstruct{pmix_info_t} structure. Allocation and manipulation of the list can then be accomplished using existing standard methods. Upon completion, the final \refarg{info} array can be allocated based on the number of elements on the list, and then the values in the list object \refstruct{pmix_info_t} structures transferred to the corresponding array element utilizing the \refmacro{PMIX_INFO_XFER} macro.
+\advicermend
+
+\label{cptr:api_server:noderegex}A common building block used in several areas is the construction of a regular expression identifying the nodes involved in that area - e.g., the nodes in a \refterm{session} or \refterm{job}. \ac{PMIx} provides several tools to facilitate this operation, beginning by constructing an argv-like array of node names. This array is then passed to the \refapi{PMIx_generate_regex} function to create a regular expression parseable by the \ac{PMIx} server library, as shown below:
+
+\cspecificstart
+\begin{codepar}
+char **nodes = NULL;
+char *nodelist;
+char *regex;
+size_t n;
+pmix_status_t rc;
+pmix_info_t info;
+
+/* loop over an array of nodes, adding each
+ * name to the array */
+for (n=0; n < num_nodes; n++) {
+    PMIX_ARGV_APPEND(&nodes, node[n]->name);
```

Comment:
I've found that people tend to copy/paste these snippets and expect them to compile. Hence, I'd prefer to keep it so it can compile, but have added a comment about filtering.